generator/src/generator/namespace.rs

```diff
@@ -3404,15 +3482,27 @@ impl<'ns, 'c> NamespaceGenerator<'ns, 'c> {
                                 true,
                             ),
                         );
-                        outln!(
-                            out,
-                            "let {} = {}.to_vec();",
-                            rust_field_name,
-                            rust_field_name
-                        );
+                        // Only force taking ownership for non-request types.
+                        if container == FieldContainer::Other {
+                            outln!(
+                                out,
+                                "let {} = {}.to_vec();",
+                                rust_field_name,
+                                rust_field_name
+                            );
+                        }
                     } else {
-                        outln!(out, "let {} = remaining.to_vec();", rust_field_name);
-                        outln!(out, "let remaining = &remaining[remaining.len()..];");
+                        // Only force taking ownership for non-request types.
```

Comment:
(unrelated to this PR) What would it take to make "normal struct parsing" also be able to borrow stuff from the input? The hand-written API moved away from this being possible, but how close is the code generator?

Reply:
I'd have to look at some of the generated code but I don't think it would be particularly difficult at this point.